example-backend/src/main/kotlin/app/truid/example/examplebackend/ExampleBackendApplication.kt

```diff
@@ -30,7 +33,12 @@ class ExampleBackendApplication {
 
     @Bean
     fun webClient(): WebClient {
-        return WebClient.create()
+        return WebClient
+            .builder()
+            // create a new connection for each request to avoid connection reset by peer when docker compose container
+            // is restarted
```

Comment:
Docker compose container is truid backend? The audience for the example code doesn't have one of those and will not understand what this comment is.

Is there a connection pool that doesn't realize when a connection is reset? Doesn't the docker containers close all sockets when terminating?

Maybe easier to configure the client not to use a connection pool?

Reply:
Docker compose container is example backend, which is part of how to run the application. 
Not sure what is causing the connection reset by peer, but one reproducable way is to rebuild and restart the web example. Or the example backend. I.e. not restart the full docker compose. 

So my guess it is something related to how docker compose sets up the network..